apollo-ios/Sources/Apollo/IncrementalGraphQLResponse.swift

```diff
@@ -40,9 +40,72 @@ final class IncrementalGraphQLResponse<Operation: GraphQLOperation> {
     )
   }
 
-  func parseIncrementalResult() throws -> IncrementalGraphQLResult {
-    guard let path = base.body["path"] as? [JSONValue] else { throw ResponseError.missingPath }
-    guard let label = base.body["label"] as? String else { throw ResponseError.missingLabel }
+  /// Parses the response into a `IncrementalGraphQLResult` and a `RecordSet` depending on the cache policy. The result
+  /// can be used to merge into a partial result and the `RecordSet` can be merged into a local cache.
+  ///
+  /// - Returns: A tuple of a `IncrementalGraphQLResult` and an optional `RecordSet`.
+  ///
+  /// - Parameter cachePolicy: Used to determine whether a cache `RecordSet` is returned. A cache policy that does
+  /// not read or write to the cache will return a `nil` cache `RecordSet`.
+  func parseIncrementalResult(
+    withCachePolicy cachePolicy: CachePolicy
+  ) throws -> (IncrementalGraphQLResult, RecordSet?) {
+    switch cachePolicy {
+    case .fetchIgnoringCacheCompletely:
+      // There is no cache, so we don't need to get any info on dependencies. Use fast parsing.
+      return (try parseIncrementalResultFast(), nil)
+
+    default:
+      return try parseIncrementalResult()
+    }
+  }
+
+  private func parseIncrementalResult() throws -> (IncrementalGraphQLResult, RecordSet?) {
```

Comment:
Ideally the `parseResult` and `parseResultFast` functions in GraphQLResult would be `internal` rather than the `public` scope they have now but that would be a potentially breaking change. These are scoped minimally for now but we can always loosen that if needed.